File: backEnd/src/main/java/com/quiz/ourclass/domain/challenge/dto/response/RunningChallengeResponse.java
package com.quiz.ourclass.domain.challenge.dto.response;

import com.quiz.ourclass.domain.challenge.dto.ChallengeSimpleDTO;
import lombok.Builder;

@Builder
public record RunningChallengeResponse(
    ChallengeSimpleDTO challengeSimpleDTO,
    int waitingCount,
    int doneMemberCount
) {

}

File: android/data/src/main/java/com/sixkids/data/model/response/RunningChallengeResponse.kt
package com.sixkids.data.model.response

import com.sixkids.model.RunningChallenge
import com.squareup.moshi.Json

data class RunningChallengeResponse(
    @Json(name = "challengeSimpleDTO")
    val challenge: ChallengeResponse,
    val waitingCount: Int,
    val doneMemberCount: Int
)

internal fun RunningChallengeResponse.toModel(): RunningChallenge {
    val challenge = challenge.toModel()
    return RunningChallenge(
        id = challenge.id,
        title = challenge.title,
        content = challenge.content,
        totalMemberCount = challenge.headCount,
        doneMemberCount = doneMemberCount,
        startTime = challenge.startTime,
        endTime = challenge.endTime,
        waitingCount = waitingCount
    )
}
